tests/nopanic.ci

```diff
@@ -9,14 +13,12 @@ print("Files to check: %s" % srcs)
 
 failed = False
 for f in srcs:
-    print("Checking file %s" % f)
-    contents = open("src/%s" % f, "r").read().split("#[cfg(test)]")[0]
-    for b in banned:
-        if b not in contents:
-            continue
-
-        failed = True
-        print("File %s calls banned function: %s)" % (f, b))
-    pass
-
+    with open("src/" + f) as src_file:
+        for line_no, line in enumerate(src_file):
+            for b in banned:
+                if b not in line or "#[allow_ci]" in line:
```

Comment:
I think this would need to check for `#[allow_ci]` in the previous line, unless I am misreading this.

Reply:
Hmm, this seems to work well:

https://github.com/keylime/rust-keylime/runs/1413393155?check_suite_focus=true

The previous line is just to go over the banned dict , I think this could likely be optimized a bit better using regexp, but we don't need to worry about performance too much

Reply:
it looks like it was not even working before (it was allowing a load of unwraps).

Reply:
Ah, yeah, that's a lot of unwraps - it's good to have this as an improvement. 

I am seeing the line numbers reported as one off from where the `unwrap()` is called, though, ex. `File crypto.rs on line number  200 calls banned function: unwrap()` while I see the `unwrap()` on 201: https://github.com/keylime/rust-keylime/blob/master/src/crypto.rs#L201. Not a big deal but easy to change.

Reply:
My concern is more about the `#[allow_ci]` being placed on the previous line from `panic!` or `unwrap()` - would that be caught here?

Reply:
I see now, so I planned to have it on the same line as the instance of panic! / unwrap

`key.unwrap() #[allow_ci]`

Here is an example of it working:

![image](https://user-images.githubusercontent.com/7058938/99512468-63fd5980-2981-11eb-9198-24bd1fd074f0.png)

(Note, this is before you're change to increment the line by +1 so it records the correct line number).

Reply:
Ah, this makes much more sense now. Thanks for the clarification!

Reply:
no worries. what you would you like to do around the naming convention? I went with allow_ci rather than a mention of panics (as it also covers unwraps as well)?  I am open to suggestions here.

Reply:
I think `allow_ci` is fine for now; if we end up allowing more types of banned calls, we may want to differentiate later. Alternatively `allow_banned` is possibly a little more descriptive.